src/Core/Manifest/ClassManifest.php

```diff
@@ -489,23 +583,57 @@ public function handleFile($basename, $pathname, $includeTests)
      */
     protected function coalesceDescendants($class)
     {
-        $lClass = strtolower($class);
+        // Reset descendents to immediate children initially
+        $lowerClass = strtolower($class);
+        if (empty($this->children[$lowerClass])) {
+            return [];
+        }
 
-        if (array_key_exists($lClass, $this->children)) {
-            $this->descendants[$lClass] = array();
+        // Coalasce children into descendent list
+        $this->descendants[$lowerClass] = $this->children[$lowerClass];
+        foreach ($this->children[$lowerClass] as $childClass) {
+            // Merge all nested descendants
+            $this->descendants[$lowerClass] = array_merge(
+                $this->descendants[$lowerClass],
+                $this->coalesceDescendants($childClass)
+            );
+        }
+        return $this->descendants[$lowerClass];
+    }
 
-            foreach ($this->children[$lClass] as $class) {
-                $this->descendants[$lClass] = array_merge(
-                    $this->descendants[$lClass],
-                    array($class),
-                    $this->coalesceDescendants($class)
-                );
+    /**
+     * Reload state from given cache data
+     *
+     * @param array $data
+     * @return bool True if cache was valid and successfully loaded
+     */
+    protected function loadFromCache($data)
```

Comment:
what about naming this `loadToState`?
I was confused with the current naming, as "from" implies you're returning data from the cache, as opposed to pushing data to the state
This might be because I wasn't thinking about the cache being external...